client/landing/jetpack-cloud/components/filter-bar/activity-type-selector/index.tsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import { useTranslate } from 'i18n-calypso';
+import React, { FunctionComponent, useRef, ChangeEvent, useEffect, useState } from 'react';
+import { isEqual } from 'lodash';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { Button } from '@automattic/components';
+import FormCheckbox from 'components/forms/form-checkbox';
+import FormLabel from 'components/forms/form-label';
+import Popover from 'components/popover';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+interface ActivityCount {
+	count: number;
+	key: string;
+	name: string;
+}
+
+interface Props {
+	groups: string[];
+	onGroupsChange: ( newGroups: string[] ) => void;
+	isVisible: boolean;
+	onClick: () => void;
+	onClose: () => void;
+	activityCounts: ActivityCount[];
+}
+
+const ActivityTypeSelector: FunctionComponent< Props > = ( {
+	isVisible,
+	onClick,
+	onClose,
+	activityCounts,
+	groups,
+	onGroupsChange,
+} ) => {
+	const translate = useTranslate();
+	const buttonRef = useRef< Button >();
+
+	const [ localGroups, setLocalGroups ] = useState( groups );
+
+	const onChange = ( { target: { checked, name } }: ChangeEvent< HTMLInputElement > ) => {
+		setLocalGroups(
+			( checked ? [ ...localGroups, name ] : localGroups.filter( ( key ) => key !== name ) ).filter(
+				( value, index, self ) => self.indexOf( value ) === index
+			)
+		);
+	};
+
+	const onApply = () => {
+		onGroupsChange( localGroups );
+		onClose();
+	};
+
+	const onSelectAll = () => {
+		setLocalGroups( activityCounts.map( ( { key } ) => key ) );
+	};
+
+	const hasChanges = ! isEqual( groups, localGroups );
+
+	const hasAllSelected = localGroups.length === activityCounts.length;
+
+	// whenever we get a new controlled value or we are hidden/visible that is the new standard
+	useEffect( () => {
+		setLocalGroups( groups );
+	}, [ groups, isVisible ] );
+
+	return (
+		<div className="activity-type-selector">
+			<Button
+				className={
+					isVisible
+						? 'activity-type-selector__main-button-active'
+						: 'activity-type-selector__main-button'
+				}
+				compact
+				onClick={ onClick }
+				ref={ buttonRef }
+			>
+				{ translate( 'Activity type' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **24** times:
`translate( 'Activity Type' )` *ES Score: 10*